### PR TITLE
API: added OpenAPI document 2nd PR

### PR DESCRIPTION
**Pull request by juliuskoskela**
Moved stalled PR https://github.com/tiiuae/vmd/pull/24 from a fork into a branch inside the repository. This was done so more team members could make commits to the PR.

## Open items from previous PR

- [x] https://github.com/tiiuae/vmd/pull/24#discussion_r1142197766
- [x] https://github.com/tiiuae/vmd/pull/24#discussion_r1142179683
- [x] https://github.com/tiiuae/vmd/pull/24#discussion_r1142205650
- [x] https://github.com/tiiuae/vmd/pull/24#discussion_r1142210296
- [x] https://github.com/tiiuae/vmd/pull/24#discussion_r1142212084

### CODE REVIEW

File: vmd-api/api/openapi.yaml [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+        emulator: /usr/bin/qemu-system-x86_64
+        hypervisor: KVM
+        arch: aarch64
+        chipset: Q35
+        firmware: BIOS
+      properties:
+        hypervisor:
+          enum:
+          - KVM
+          - Xen
```
Comment (vilvo) [inline vmd-api/api/openapi.yaml:230]: We won't support `Xen`. It's misleading and should be removed but is not a blocker.

**Review by vilvo — APPROVED** [review]

File: vmd-api/api/openapi.yaml (hunk 1 above)
Comment (juliuskoskela) [reply vmd-api/api/openapi.yaml:230]: I will note this and move it to the next PR

Comment (juliuskoskela) [reply vmd-api/api/openapi.yaml:230]: Resolved in https://github.com/tiiuae/vmd/pull/35/commits/751bf9b99f2cff749e816f0ae279fd2167af8915

**Comment by vilvo** [comment]
@baz2142 's commit before @juliuskoskela 's commits can't be signed anymore without rewriting the history. I'll merge this now bypassing branch protections.

**Pull request merged**